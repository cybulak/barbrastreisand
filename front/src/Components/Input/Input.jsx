import React from 'react';
import PropTypes from 'prop-types';
import { Input as SemanticInput, Message } from 'semantic-ui-react';

export const Input = ({
	icon,
	iconPosition,
	placeholder,
	value,
	onChange,
	name,
	validation
}) => (
	<div className="input-container">
		<SemanticInput
			icon={icon}
			iconPosition={iconPosition}
			placeholder={placeholder}
			value={value}
			onChange={onChange}
			name={name}
			error={validation && validation.isInvalid}
		/>
		{validation && validation.isInvalid && (
			<Message error header={`Invalid ${name}`} content={validation.message} />
		)}
	</div>
);

Input.propTypes = {
	icon: PropTypes.string,
	iconPosition: PropTypes.oneOf(['left', 'right']),
	placeholder: PropTypes.string.isRequired,
	value: PropTypes.string.isRequired,
	onChange: PropTypes.func.isRequired,
	name: PropTypes.string.isRequired,
	validation: PropTypes.oneOfType([PropTypes.object, PropTypes.bool])
};

Input.defaultProps = {
	icon: '',
	iconPosition: 'left',
	validation: false
};
