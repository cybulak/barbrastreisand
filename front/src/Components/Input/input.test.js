import React from 'react';
import renderer from 'react-test-renderer';
import { Input } from './Input';

it('render correctly input component', () => {
	const input = renderer
		.create(
			<Input placeholder="test" value="test" onChange={() => {}} name="test" />
		)
		.toJSON();
	expect(input).toMatchSnapshot();
});
