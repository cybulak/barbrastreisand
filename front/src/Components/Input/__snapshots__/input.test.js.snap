// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly input component 1`] = `
<div
  className="input-container"
>
  <div
    className="ui left icon input"
  >
    <input
      name="test"
      onChange={[Function]}
      placeholder="test"
      type="text"
      value="test"
    />
    <i
      aria-hidden="true"
      className="icon"
      onClick={[Function]}
    />
  </div>
</div>
`;
